src/spatial_reaction_systems/spatial_reactions.jl

```diff
@@ -0,0 +1,96 @@
+### Spatial Reaction Structures ###
+
+# Abstract spatial reaction structures.
+abstract type AbstractSpatialReaction end
+
+### EdgeParameter Metadata ###
+
+# Implements the edgeparameter metadata field.
+struct EdgeParameter end
+Symbolics.option_to_metadata_type(::Val{:edgeparameter}) = EdgeParameter
+
+# Implements the isedgeparameter check function.
+isedgeparameter(x::Num, args...) = isedgeparameter(Symbolics.unwrap(x), args...)
+function isedgeparameter(x, default = false)
+    p = Symbolics.getparent(x, nothing)
+    p === nothing || (x = p)
+    Symbolics.getmetadata(x, EdgeParameter, default)
+end
+
+### Transport Reaction Structures ###
+
+# A transport reaction. These are simple to hanlde, and should cover most types of spatial reactions.
+# Only permit constant rates (possibly consisting of several parameters).
+struct TransportReaction <: AbstractSpatialReaction
+    """The rate function (excluding mass action terms). Currently only constants supported"""
+    rate::Num
+    """The species that is subject to difusion."""
+    species::BasicSymbolic{Real}
+
+    # Creates a diffusion reaction.
+    function TransportReaction(rate::Num, species::Num)
+        new(rate, species.val)
+    end
+end
+# If the rate is a value, covert that to a numemric expression.
+function TransportReaction(rate::Number, args...)
+    TransportReaction(Num(rate), args...)
+end
```

Comment:
See later comment on storing using `Num` (basically, these maybe be equations). The `Number` use was bad though, will fix.

Reply:
See preceding comment. Always unwrap `Num`s in library code.

Reply:
Sorry, got confused about what unwrapping meant. Thought you referred to our previous discussion where I used `Num` instead of `BasicSymbolic{Real}` and you wanted the same here. Will change to `Any`.

Reply:
fixed